vertx-mssql-client/src/main/java/io/vertx/mssqlclient/impl/codec/ExtendedBatchQueryCommandCodec.java

```diff
@@ -26,22 +26,22 @@
   private int paramsIdx;
   private int messageDecoded;
 
-  ExtendedBatchQueryCommandCodec(ExtendedQueryCommand<T> cmd) {
-    super(cmd);
+  ExtendedBatchQueryCommandCodec(TdsMessageCodec tdsMessageCodec, ExtendedQueryCommand<T> cmd) {
+    super(tdsMessageCodec, cmd);
     paramsList = cmd.paramsList();
   }
 
   @Override
-  void encode(TdsMessageEncoder encoder) {
+  void encode() {
     if (paramsList.isEmpty()) {
       completionHandler.handle(CommandResponse.failure("Can not execute batch query with 0 sets of batch parameters."));
       return;
     }
-    super.encode(encoder);
+    super.encode();
   }
 
   @Override
-  protected void handleMessageDecoded() {
+  protected void handleDecodingComplete() {
     if (paramsList.size() == 1 || ++messageDecoded == 2) {
```

Comment:
++ will only be applied when `paramsList.size() != -1` is that the correct thing that is to do here ? i.e is it equivalent to

```java
if (paramsList.size() == 1) {
   ...
} else {
  messageDecoded++;
  if (messageDecoded == 2) {
    ...
  } else {
    ...
  }
}
```

Reply:
it is not part of this review but a general remark when seing this

Reply:
Yes, it is on purpose and equivalent to the snippet above.